Add the Orleans meters (#36931)
dotnet/docs: docs/orleans/host/monitoring/index.md

For more information, see [Investigate performance counters (dotnet-counters)](../../../core/diagnostics/dotnet-counters.md).

### Orleans meters

Orleans uses the [System.Diagnostics.Metrics](../../../core/diagnostics/compare-metric-apis.md#systemdiagnosticsmetrics) APIs to collect metrics. Orleans categorizes each meter into domain-centric concerns, such as networking, messaging, gateway, and so on. The following subsections describe the meters that Orleans uses.

#### Networking

> A collection of networking meters that are used to monitor the networking layer.

Author: ```suggestion
The following table represents a collection of networking meters that are used to monitor the Orleans networking layer.
```

| `orleans-messaging-pings-sent` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of pings sent. |
| `orleans-messaging-pings-received` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of pings received. |
| `orleans-messaging-pings-reply-received` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of ping replies received. |
| `orleans-messaging-pings-reply-missed` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of ping replies missed. |
| `orleans-messaging-expired"` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of messages that have expired. |
| `orleans-messaging-rejected` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of messages that have been rejected. |
| `orleans-messaging-rerouted` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of messages that have been rerouted. |
| `orleans-messaging-sent-local` | <xref:System.Diagnostics.Metrics.ObservableCounter%601> | An observable counter representing the number of local messages sent. |

#### Gateway

> A collection of gateway meters that are used to monitor the gateway layer.

Author: ```suggestion
The following table represents a collection of gateway meters that are used to monitor the Orleans gateway layer.
```

| `orleans-runtime-available-memory` | <xref:System.Diagnostics.Metrics.ObservableCounter%601> | An observable counter representing the available memory (in MB) for the Orleans runtime. |

#### Catalog

A collection of catalog meters that are used to monitor the catalog layer.

| Meter name | Type | Description |
|--|--|--|
| `orleans-catalog-activations` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the number of catalog activations. |
| `orleans-catalog-activation-working-set` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the number of activations within the working set. |
| `orleans-catalog-activation-created` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of created activations. |
| `orleans-catalog-activation-destroyed` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of destroyed activations. |
| `orleans-catalog-activation-failed-to-activate` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of activations that failed to activate. |

> | `orleans-catalog-activation-collections` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of activation collections. |

Reviewer: ```suggestion
| `orleans-catalog-activation-collections` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of idle activation collections. |
```

| `orleans-catalog-activation-shutdown` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of shutdown activations. |
| `orleans-catalog-activation-non-existent` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of non-existent activations. |
| `orleans-catalog-activation-concurrent-registration-attempts` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of concurrent activation registration attempts. |

#### Directory

> A collection of directory meters that are used to monitor the directory layer.

Author: ```suggestion
The following table represents a collection of directory meters that are used to monitor the Orleans directory layer.
```

| `orleans-directory-lookups-local-directory-successes` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of local directory successful lookups. |
| `orleans-directory-lookups-cache-issued` | <xref:System.Diagnostics.Metrics.Counter%601> | A count cached lookups issued. |
| `orleans-directory-lookups-cache-successes` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of cached successful lookups. |
| `orleans-directory-validations-cache-sent` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of directory cache validations sent.  |
| `orleans-directory-validations-cache-received` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of directory cache validations received. |
| `orleans-directory-partition-size` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the directory partition size. |
| `orleans-directory-cache-size` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the directory cache size. |
| `orleans-directory-ring-size` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the directory ring size. |

> | `orleans-directory-ring-local-portion-distance` |<xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the directories local portion distance. |
> | `orleans-directory-ring-local-portion-percentage` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the directories local portion percentage. |
> | `orleans-directory-ring-local-portion-average-percentage` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the directories local average percentage. |

Reviewer: ```suggestion
| `orleans-directory-ring-local-portion-distance` |<xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the ring range owned by the local directory partition. |
| `orleans-directory-ring-local-portion-percentage` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the ring range owned by the local directory, represented as a percentage of the total range. |
| `orleans-directory-ring-local-portion-average-percentage` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the average percentage of the directory ring range owned by each silo, giving a representation of how balanced directory ownership. |
```

> | `orleans-directory-registrations-single-act-issued` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of directory single act registrations issued. |
> | `orleans-directory-registrations-single-act-local` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of directory single act registrations local. |
> | `orleans-directory-registrations-single-act-remote-sent` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of directory single act registrations sent remotely. |
> | `orleans-directory-registrations-single-act-remote-received` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of directory single act registrations received remotely. |

Reviewer: ```suggestion
| `orleans-directory-registrations-single-act-issued` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of directory single activation registrations issued. |
| `orleans-directory-registrations-single-act-local` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of directory single activation registrations handled by the local directory partition. |
| `orleans-directory-registrations-single-act-remote-sent` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of directory single activation registrations sent to a remote directory partition. |
| `orleans-directory-registrations-single-act-remote-received` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of directory single activation registrations received from remote hosts. |
```

| `orleans-directory-unregistrations-local` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of directory deregistrations handled by the local directory partition. |
| `orleans-directory-unregistrations-remote-sent` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of directory deregistrations sent to remote directory partitions. |
| `orleans-directory-unregistrations-remote-received` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of directory deregistrations received from remote hosts. |
| `orleans-directory-unregistrations-many-issued` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of directory multi-activation deregistrations issued. |
| `orleans-directory-unregistrations-many-remote-sent` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of directory multi-activations deregistrations sent to remote directory partitions. |
| `orleans-directory-unregistrations-many-remote-received` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of directory multi-activation deregistrations received from remote hosts. |

#### Consistent ring

> A collection of consistent ring meters that are used to monitor the consistent ring layer.

Author: ```suggestion
The following table represents a collection of consistent ring meters that are used to monitor the Orleans consistent ring layer.
```

| Meter name | Type | Description |
|--|--|--|
| `orleans-consistent-ring-size` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the consistent ring size. |
| `orleans-consistent-ring-range-percentage-local` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the consistent ring local percentage. |
| `orleans-consistent-ring-range-percentage-average` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the consistent ring average percentage. |

#### Watchdog

> A collection of watchdog meters that are used to monitor the watchdog layer.

Author: ```suggestion
The following table represents a collection of watchdog meters that are used to monitor the Orleans watchdog layer.
```

| Meter name | Type | Description |
|--|--|--|
| `orleans-watchdog-health-checks` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of watchdog health checks. |
| `orleans-watchdog-health-checks-failed` | <xref:System.Diagnostics.Metrics.Counter%601> | A count of failed watchdog health checks. |

#### Client

> A collection of client meters that are used to monitor the client layer.

Author: ```suggestion
The following table represents a collection of client meters that are used to monitor the Orleans client layer.
```

| Meter name | Type | Description |
|--|--|--|
| `orleans-client-connected-gateways` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the number of connected gateway clients. |

#### Miscellaneous

A collection of miscellaneous meters that are used to monitor various layers.

| Meter name | Type | Description |
|--|--|--|
| `orleans-grains` | <xref:System.Diagnostics.Metrics.Counter%601> | A count representing the number of grains. |
| `orleans-system-targets` | <xref:System.Diagnostics.Metrics.Counter%601> | A count representing the number of system targets. |

#### App requests

> A collection of app request meters that are used to monitor the app request layer.

Author: ```suggestion
The following table represents a collection of app request meters that are used to monitor the Orleans app request layer.
```

| Meter name | Type | Description |
|--|--|--|
| `orleans-app-requests-latency` | <xref:System.Diagnostics.Metrics.ObservableCounter%601> | An observable counter representing app request latency. |
| `orleans-app-requests-timedout` | <xref:System.Diagnostics.Metrics.ObservableCounter%601> |  An observable counter representing app requests that have timed out. |

#### Reminders

A collection of reminder meters that are used to monitor the reminder layer.

| Meter name | Type | Description |
|--|--|--|
| `orleans-reminders-tardiness` | <xref:System.Diagnostics.Metrics.Histogram%601> | A histogram representing the number of seconds a reminder is tardy. |
| `orleans-reminders-active` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the number active reminders. |
| `orleans-reminders-ticks-delivered` | <xref:System.Diagnostics.Metrics.Counter%601> | A count representing the number of reminder ticks that have been delivered. |

#### Storage

> A collection of storage meters that are used to monitor the storage layer.

Author: ```suggestion
The following table represents a collection of storage meters that are used to monitor the Orleans storage layer.
```

| Meter name | Type | Description |
|--|--|--|
| `orleans-storage-read-errors` | <xref:System.Diagnostics.Metrics.Counter%601> |  A count representing the number of storage read errors. |
| `orleans-storage-write-errors` | <xref:System.Diagnostics.Metrics.Counter%601> |  A count representing the number of storage write errors. |
| `orleans-storage-clear-errors` | <xref:System.Diagnostics.Metrics.Counter%601> |  A count representing the number of storage clear errors. |
| `orleans-storage-read-latency` | <xref:System.Diagnostics.Metrics.Histogram%601> | A histogram representing the storage read latency in milliseconds. |
| `orleans-storage-write-latency` | <xref:System.Diagnostics.Metrics.Histogram%601> | A histogram representing the storage write latency in milliseconds. |
| `orleans-storage-clear-latency` | <xref:System.Diagnostics.Metrics.Histogram%601> | A histogram representing the storage clear latency in milliseconds. |

#### Streams

> A collection of stream meters that are used to monitor the stream layer.

Author: ```suggestion
The following table represents a collection of stream meters that are used to monitor the Orleans stream layer.
```

| `orleans-streams-queue-cache-memory-released` | <xref:System.Diagnostics.Metrics.ObservableCounter%601> | An observable counter representing the stream queue memory released. |
| `orleans-streams-queue-cache-oldest-to-newest-duration` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the duration from the oldest to the newest stream queue cache. |
| `orleans-streams-queue-cache-oldest-age` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the age of the oldest cached message. |
| `orleans-streams-queue-cache-pressure` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing the pressure on the stream queue cache. |
| `orleans-streams-queue-cache-under-pressure` | <xref:System.Diagnostics.Metrics.ObservableGauge%601> | An observable gauge representing whether the stream queue cache is under pressure. |
| `orleans-streams-queue-cache-pressure-contribution-count` | <xref:System.Diagnostics.Metrics.ObservableCounter%601> | An observable counter representing the stream queue cache pressure contributions. |

#### Transactions

> A collection of transaction meters that are used to monitor the transaction layer.

Author: ```suggestion
The following table represents a collection of transaction meters that are used to monitor the Orleans transaction layer.
```
